File: backend/ResearchCruiseApp/Application/Models/DTOs/CruiseApplications/FormCDto.cs
using System.ComponentModel.DataAnnotations;

namespace ResearchCruiseApp.Application.Models.DTOs.CruiseApplications;

public class FormCDto
{
    [StringLength(1)]
    public string ShipUsage { get; init; } = null!;

    [StringLength(1024)]
    public string DifferentUsage { get; init; } = null!;

    public List<PermissionDto> Permissions { get; init; } = [];

    public List<ResearchAreaDescriptionDto> ResearchAreaDescriptions { get; init; } = [];

    public List<UgTeamDto> UgTeams { get; init; } = [];

    public List<GuestTeamDto> GuestTeams { get; init; } = [];

    public List<ResearchTaskEffectDto> ResearchTasksEffects { get; init; } = [];

    public List<ContractDto> Contracts { get; init; } = [];

    public List<SpubTaskDto> SpubTasks { get; init; } = [];

    public List<ShortResearchEquipmentDto> ShortResearchEquipments { get; init; } = [];

    public List<LongResearchEquipmentDto> LongResearchEquipments { get; init; } = [];

    public List<PortDto> Ports { get; init; } = [];

    public List<CruiseDayDetailsDto> CruiseDaysDetails { get; init; } = [];

    public List<ResearchEquipmentDto> ResearchEquipments { get; init; } = [];

    public List<Guid> ShipEquipmentsIds { get; init; } = [];

    public List<CollectedSampleDto> CollectedSamples { get; init; } = [];

    [StringLength(10240)]
    public string? SpubReportData { get; init; } = null!;

    [StringLength(10240)]
    public string? AdditionalDescription { get; init; } = null!;

    public List<FileDto> Photos { get; init; } = []!;
}

File: frontend/src/modules/cruise-applications/models/FormCDto.ts
import { CollectedSampleDto } from '@/cruise-applications/models/CollectedSampleDto';
import { ContractDto } from '@/cruise-applications/models/ContractDto';
import { CruiseDayDetailsDto } from '@/cruise-applications/models/CruiseDayDetailsDto';
import { FileDto } from '@/cruise-applications/models/FileDto';
import { GuestTeamDto } from '@/cruise-applications/models/GuestTeamDto';
import { LongResearchEquipmentDto } from '@/cruise-applications/models/LongResearchEquipmentDto';
import { PermissionDto } from '@/cruise-applications/models/PermissionDto';
import { PortDto } from '@/cruise-applications/models/PortDto';
import { ResearchAreaDescriptionDto } from '@/cruise-applications/models/ResearchAreaDescriptionDto';
import { ResearchEquipmentDto } from '@/cruise-applications/models/ResearchEquipmentDto';
import { ResearchTaskEffectDto } from '@/cruise-applications/models/ResearchTaskEffectDto';
import { ShortResearchEquipmentDto } from '@/cruise-applications/models/ShortResearchEquipmentDto';
import { SpubTaskDto } from '@/cruise-applications/models/SpubTaskDto';
import { UGTeamDto } from '@/cruise-applications/models/UGTeamDto';

export type FormCDto = {
  shipUsage: string; // Max length 1
  differentUsage: string; // Max length 1024
  permissions: PermissionDto[];
  researchAreaDescriptions: ResearchAreaDescriptionDto[];
  ugTeams: UGTeamDto[];
  guestTeams: GuestTeamDto[];
  researchTasksEffects: ResearchTaskEffectDto[];
  contracts: ContractDto[];
  spubTasks: SpubTaskDto[];
  shortResearchEquipments: ShortResearchEquipmentDto[];
  longResearchEquipments: LongResearchEquipmentDto[];
  ports: PortDto[];
  cruiseDaysDetails: CruiseDayDetailsDto[];
  researchEquipments: ResearchEquipmentDto[];
  shipEquipmentsIds: string[];
  collectedSamples: CollectedSampleDto[];
  spubReportData?: string; // Max length 10240
  additionalDescription?: string; // Max length 10240
  photos: FileDto[];
};
